authentication/src/main/kotlin/com/uber/sdk2/auth/client/UberAuthClient.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2024. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.uber.sdk2.auth.client
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import androidx.activity.result.ActivityResultLauncher
+import com.uber.sdk2.auth.AuthActivity
+import com.uber.sdk2.auth.api.request.AuthContext
+
+/**
+ * Client to authenticate the user against one of the available Uber apps. It is entrypoint for the
+ * consumers to initiate the authentication flow
+ */
+object UberAuthClient {
```

Comment:
Do we want this to be an `interface` so we can provide alternative impl? (example: tests)

Reply:
The rationale behind object was to provide a single yet easy way to launch the auth flow. 
But, your raise a good point about testability. I'll update with a new patch